torch/csrc/jit/passes/onnx/fixup_jit_exceptions.cpp

```diff
@@ -0,0 +1,89 @@
+#include <torch/csrc/jit/passes/onnx/fixup_jit_exceptions.h>
+
+namespace torch {
+namespace jit {
+
+namespace onnx{
+using namespace ::c10::onnx;
+}
+
+bool IsExceptionBlock(Block* block) {
+  for (auto* output : block->outputs()) {
+    if (output->node()->kind() == ::c10::prim::Uninitialized) {
+      return true;
+    }
+  }
+  return false;
+}
+void RemoveUninitialized(Block* block) {
+  std::vector<Node*> to_destroy;
+  for (auto* node : block->nodes()) {
+    if (node->kind() == ::c10::prim::Uninitialized) {
```

Comment:
Is it safe to assume that no other nodes would be consuming `prim::Uninitialized` node as input?

Reply:
This is a good question. A related/similar question is whether the exception scenario is the only scenario when prim::Uninitialized comes into play.

Reply:
I have not been able to find any case in which Unititialized is used except for an exception. If there is one I am not aware of it, and it would seem like it would be a very poorly created model if it did exist.

Reply:
@houseroad Are you aware of any other case in which a prim::Unititialized would be used, I can only find cases for an exception.

Reply:
@spandantiwari I was able to find another case in which we can generate a `prim::Uninitialized`. However the only cases in which I can generate it outside of an exception are rather contrived examples that would be better to simply rewrite the model which can easily be done, then try to handle in a pass. For example the following forward function generates a `prim::Uninitialized`. Though do note that currently this pass function would not modify the graph.
```python
def forward(self, y):
    z = y + 1
    if y.size(0) < 6:
        y += 4
    else:
        return y
    return y
```
This generates the following graph prior to my cleanup pass
```
graph(%y.1 : Float(3, 4, 5)):
  %16 : Long() = prim::Constant[value={1}]()
  %17 : Long() = prim::Constant[value={0}]()
  %18 : Long() = prim::Constant[value={6}]()
  %19 : Long() = prim::Constant[value={4}]()
  %20 : Long() = prim::Constant[value={0}]()
  %21 : Long() = prim::Constant[value={1}]()
  %7 : Tensor = prim::Uninitialized()
  %8 : Long() = aten::size(%y.1, %17) 
  %9 : Long() = aten::lt(%8, %18) 
  %10 : Long(), %11 : Tensor, %y : Tensor = prim::If(%9) 
    block0():
      %y.5 : Tensor = aten::add(%y.1, %19, %16)
      -> (%20, %7, %y.5)
    block1():
      -> (%21, %y.1, %7)
  %14 : Tensor = prim::If(%10)
    block0():
      -> (%11)
    block1():
      -> (%y)
  return (%14)
```
Because the use of the `prim::Uninitialized` is within both blocks of the code, my pass would skip that and never modify it. As such it would still fail at symbolics registry. However this forward function could easily be re-written to remove the issue entirely. Based on what I can see from the usage of `prim::Uninitialized` the only time it would be caught by my pass detector is when it is an exception. In the cases where it is not caught I.E. early return like above the model code itself can easily be updated to not generate a `prim::Uninitialized`.